spec/extended_git/annex_spec.rb

```diff
@@ -0,0 +1,299 @@
+      temp.annex.sync(content: true)
+      temp.annex.uninit
+      FileUtils.rm_r(temp.dir.path)
+    end
+  end
+
+  shared_context 'add directory special remote' do
+    let(:special_remote_name) { 'local_directory' }
+    let(:special_remote_dir) { Rails.root.join('tmp', 'test_special_remote', repo_name).to_s }
+
+    before do
+      FileUtils.mkdir_p(special_remote_dir)
+      temp = ExtendedGit.clone(remote_origin, repo_name + '_temp', path: Rails.root.join('tmp'))
+      temp.annex.init
+      temp.annex.initremote(special_remote_name, type: 'directory', directory: special_remote_dir, encryption: 'none')
+      temp.annex.sync
+      temp.annex.uninit
+      FileUtils.rm_r(temp.dir.path)
+    end
+
+    after do
+      FileUtils.chmod_R(0755, special_remote_dir)
+      FileUtils.rm_r(special_remote_dir)
+    end
+  end
+
+  before do
+    ExtendedGit.init(nil, repository: remote_origin, bare: true)
+  end
+
+  after do
+    # Remove working directory
+    git.annex.uninit if git.config('annex.uuid').present?
+    FileUtils.rm_r(working_directory)
+
+    # Remove remote origin
+    FileUtils.chmod_R(0755, remote_origin)
+    FileUtils.rm_r(remote_origin)
+  end
+
+  describe '#init' do
+    it 'initializes repository with description' do
+      git.annex.init('test_working_directory')
+      expect(git.annex.info.remote?('test_working_directory')).to be true
+    end
+
+    it 'initializes repository with git annex' do
+      git.annex.init
+      expect(git.config('annex.uuid')).not_to be nil
+    end
+  end
+
+  describe '#version' do
+    it 'returns full version output' do
+      expect(git.annex.version).to match(/^git-annex version: \d+\.\d+\.\d+$/)
+    end
+
+    it 'returns raw version output' do
+      expect(git.annex.version(raw: true)).to match(/^\d+\.\d+\.\d+$/)
+    end
+  end
+
+  describe '#add' do
+    # Here because there's a problem with `git.status`, it doesn't work
+    # when there isn't an initial commit.
+    include_context 'add readme to repository'
+
+    let(:filename) { 'new_test_file.txt' }
+    let(:filepath) { File.join(working_directory, filename) }
+
+    before do
+      git.annex.init
+      File.open(filepath, 'w') { |f| f.write("New file -- #{filename}") }
+    end
+
+    it 'adds file to be committed' do
+      git.annex.add(filename)
+      expect(git.status.added?(filename)).to be true
+    end
+
+    it 'adds entire directory to be committed' do
+      git.annex.add('.')
+      expect(git.status.added?(filename)).to be true
+    end
+  end
+
+  describe '#get' do
+    include_context 'add readme to repository'
+
+    before { git.annex.init }
+
+    it 'retrieves file from remote' do
+      expect { git.annex.get(readme) }.not_to raise_error
+      expect(File.exist?(File.join(git.dir.path, readme))).to be true
+    end
+  end
+
+  describe '#enableremote' do
+    include_context 'add readme to repository'
+    include_context 'add directory special remote'
```

Comment:
this is great - a simple way to dry up specs without having to add a bunch of helper files :rocket: